app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/PeerGroupHandshakeAsyncTask.java

```diff
@@ -0,0 +1,150 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.net.Uri;
+import android.net.wifi.p2p.WifiP2pInfo;
+import android.os.AsyncTask;
```

Comment:
I have not gone through all the comments but can we use RxJava here? Otherwise, in some time, we'll have someone opening an issue to convert all Async to Rx as we had in the past.

Reply:
@macgills @mhutti1 I can shift to using Rx but it will take some time since I haven't used it previously and I also have [this](https://github.com/kiwix/kiwix-android/pull/1276#discussion_r303831318) to fix in the code

Reply:
I think we can leave this as a challenge for later, create a ticket like "get rid of asynctasks in local_file_transfer package"

Reply:
@Aditya-Sood can you make a ticket so I can resolve this

Reply:
I actually talked with @mhutti1 about this some time ago and included Rx as a part of my GSoC work.
If you don't mind I'd like to fix this on my own - I have already read up about the basics of Rx, so fixing this will be a useful experience

Reply:
by "resolve this" I meant this conversation, would obviously love you for you to rewrite using rx but is your intention to do it in this PR?
I would like to get this PR merged sooner rather than later and I thought you opening a ticket to do it yourself later on would be better

Reply:
Got it, I'll create a new ticket for using Rx :+1: